src/stream_base.h

```diff
@@ -182,6 +184,9 @@ class StreamResource {
   Callback<AfterWriteCb> after_write_cb_;
   Callback<AllocCb> alloc_cb_;
   Callback<ReadCb> read_cb_;
+  int bytes_read_;
```

Comment:
Come to think of it, it should be `uint64_t`.  Both `int` and `size_t` can overflow after 2 or 4 GB, which isn't out of the ordinary.